src/mongo/MongoScrapbook.ts

```diff
@@ -0,0 +1,175 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { ParseTree } from 'antlr4ts/tree/ParseTree';
+import { ANTLRInputStream as InputStream } from 'antlr4ts/ANTLRInputStream';
+import { CommonTokenStream } from 'antlr4ts/CommonTokenStream';
+import * as mongoParser from './grammar/mongoParser';
+import { MongoVisitor } from './grammar/visitors';
+import { mongoLexer } from './grammar/mongoLexer';
+import * as vscodeUtil from './../utils/vscodeUtils';
+import { CosmosEditorManager } from '../CosmosEditorManager';
+import { IAzureParentNode, AzureTreeDataProvider, IActionContext } from 'vscode-azureextensionui';
+import { MongoFindResultEditor } from './editors/MongoFindResultEditor';
+import { MongoFindOneResultEditor } from './editors/MongoFindOneResultEditor';
+import { MongoCommand } from './MongoCommand';
+import { MongoDatabaseTreeItem } from './tree/MongoDatabaseTreeItem';
+
+const output = vscodeUtil.getOutputChannel();
+const notInScrapbookMessage = "You must have a MongoDB scrapbook (*.mongo) open to run a MongoDB command.";
+
+export async function executeAllCommandsFromActiveEditor(database: IAzureParentNode<MongoDatabaseTreeItem>, extensionPath, editorManager: CosmosEditorManager, tree: AzureTreeDataProvider, context: IActionContext): Promise<void> {
+	output.appendLine("Running all commands in scrapbook...")
+	let commands = getAllCommandsFromActiveEditor();
+	await executeCommands(vscode.window.activeTextEditor, database, extensionPath, editorManager, tree, context, commands);
+}
+
```

Comment:
I see that all the commands have been moved out of the class - wanted to know what the rationale is. :)

Reply:
The rational is that I don't like static classes. They're a pain to use (have to repeatedly type the class name), and also not necessary in TypeScript.  Using plain functions is simpler, and you can always use a namespace if you want to.  Eric has warmed up to this philosophy.